### PR TITLE
DM-31787: Update instructions for merging community PRs

### PR DESCRIPTION
**Pull request by leeskelvin**

### CODE REVIEW

File: work/flow.rst [hunk 1]
```diff
@@ -217,11 +217,11 @@ Community contributors can of course use personal repositories (and forks of LSS
 When an external user has created a pull request (which should be automatically checked for style using GitHub Actions), a staff member should follow these steps to help the community member merge their contribution:
 
     1. Review the community member's pull request, following our :ref:`code review guidelines<workflow-code-review>`.
-    2. Create a jira ticket (``DM-98765`` in this example) for the contribution, and a local branch of the respective repository.
+    2. Create a jira ticket (``DM-98765`` in this example) for the contribution, and create a local branch of the respective repository (without checking it out yet).
```
Comment (ktlim) [inline work/flow.rst:220]: Actually, in my testing it seems that creating the branch is not necessary; the `fetch` is sufficient.

File: work/flow.rst [hunk 2]
```diff
@@ -230,7 +230,7 @@ When an external user has created a pull request (which should be automatically
         # Assume PR #456 is the community member's PR with newly-created Jira ticket DM-98765
         git fetch origin pull/456/head:tickets/DM-98765
 
-    4. Build and test the ticket (locally and with :ref:`Jenkins CI system <workflow-testing>`) as normal, so that it can be tracked using our existing tooling. As part of this, push to your own ticket branch:
+    4. Checkout the ticket branch and build and test the ticket (locally and with :ref:`Jenkins CI system <workflow-testing>`) as normal, so that it can be tracked using our existing tooling. As part of this, push to your own ticket branch:
```
Comment (ktlim) [inline work/flow.rst:233]: Maybe this should include the`git checkout` command since we're being explicit about everything else?

**Review by ktlim — APPROVED** [review]

File: work/flow.rst (hunk 1 above)
Comment (leeskelvin) [reply work/flow.rst]: Thanks KT - I've switched around the fetch and the checkout steps to capture this, which now reads better.

File: work/flow.rst (hunk 2 above)
Comment (leeskelvin) [reply work/flow.rst]: The existing git checkout is now step #3.

File: work/flow.rst [hunk 3]
```diff
@@ -217,18 +217,18 @@ Community contributors can of course use personal repositories (and forks of LSS
...
+        git fetch origin pull/456/head:tickets/DM-98765
 
-    3. Fetch their pull request to your ticket branch:
+    3. Checkout the local branch of the respective repository.
 
     .. code-block:: bash
 
-        # Assume PR #456 is the community member's PR with newly-created Jira ticket DM-98765
-        git fetch origin pull/456/head:tickets/DM-98765
+        git checkout -b tickets/DM-98765
```
Comment (ktlim) [inline work/flow.rst:231]: Oops, this doesn't need `-b` anymore, I believe.

**Pull request merged**